src/pattern.ts

```diff
@@ -0,0 +1,16 @@
+import micromatch from "micromatch"
+
+export type Pattern = string | RegExp
+
+export function findMatch(
```

Comment:
This replaces all the `validate*` stuff we had before. We can unit test `findMatch` separately